sdks/go/pkg/beam/io/bigqueryio/bigquery.go

```diff
@@ -90,21 +90,27 @@ func Read(s beam.Scope, project, table string, t reflect.Type) beam.PCollection
 
 	// TODO(herohde) 7/13/2017: using * is probably too inefficient. We could infer
 	// a focused query from the type.
-	return query(s, project, fmt.Sprintf("SELECT * from [%v]", table), t)
+	return query(s, project, fmt.Sprintf("SELECT * from [%v]", table), t, nil)
+}
+
+// QueryOptions represents additional options for executing a query.
+type QueryOptions struct {
```

Comment:
I'm less familiar with Big Query and it's option set, but consider switching this to a variadic list of functional options ala https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis  In particular, this avoids a breaking compile change when the (admitedly few) users of this function update their beam.

It doesn't remove the need for a QueryOptions struct to serialize the options, but it does allow the callsite to not have empty options every time.

I don't feel super strongly about this as it's likely this package will likely need updating when we fix module versioning, but it's a suggestion worth discussing.

Reply:
Done.

I agree, this is a cleaner design. I couldn't think of the phrase "variadic list of functional options" to save my life.